Flipdish.Recruiting.BE/Flipdish.Recruiting.BE.OrderSender/Flipdish.Recruiting.BE.OrderSender/Controllers/OrderController.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Mail;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Flipdish.Recruiting.BE.OrderSender.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class OrderController : ControllerBase
+    {
+        private readonly ILogger<OrderController> _logger;
+
+        public OrderController(ILogger<OrderController> logger)
+        {
+            _logger = logger;
+        }
+
+        [HttpGet]
+        public string SendOrder(int orderId)
+        {
+            Order order = GetOrder(orderId);
+            decimal taxAmount = CalculateTax(order);
+            
+            // Send Email
+            string emailSubject = $"Order Received for {order.RestaurantName}";
+            string emailBody = $@"
+                OrderId: {order.OrderId}
+                Restaurant: {order.RestaurantName}
+                FoodAmount: {order.FoodAmount}
+                TipAmount: {order.TipAmount}
+                TaxAmount: {taxAmount}
+
+                Total: {order.FoodAmount + order.TipAmount + taxAmount}";
+            
+            try
+            {
+                SendEmail("interview@flipdish.com", new[] { "to@flipdish.com" }, emailSubject, emailBody);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error sending email for order: {orderId}");
+            }
+
+            return "Order Sent!";
+        }
+
+        private Order GetOrder(int orderId)
+        {
+            StreamReader sr = new StreamReader("orders.json");
+            string orderJson = sr.ReadToEnd();
+            List<Order> orders = JsonConvert.DeserializeObject<List<Order>>(orderJson);
+
+            return orders.Single(o => o.OrderId == orderId);
+        }
+        
+        private decimal CalculateTax(Order order)
```

Comment:
Maybe inline this method, as another code smell?

Reply:
I actually meant to literally inline the method -- i.e. remove the method and replace the call to it with body :)

But it's cool :)